sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/OtlpConfigUtil.java

```diff
@@ -19,6 +19,18 @@
   static final String DATA_TYPE_TRACES = "traces";
   static final String DATA_TYPE_METRICS = "metrics";
 
+  static String getOtlpProtocol(String dataType, ConfigProperties config) {
+    String protocol =
+        config.getString(String.format("otel.experimental.exporter.otlp.%s.protocol", dataType));
```

Comment:
Nit: let's just use string concatenation instead of `String.format` - here, indeed the performance implication doesn't really matter but suggest just always using string concatenation where possible to avoid the cognitive load of reasoning out the performance implication. The readability penalty is debatable anyways, i.e. concatenation looks closer to modern language's interpolation syntax

Reply:
I'm partial to the readability of `String.format`, but am more in favor of having / following a convention on these sorts of things.